### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by EudoraNiu**
Eudora Assignment2

### CODE REVIEW

**Comment by lizarova777** [comment]
Can you please upload the .Rmd file?

**Comment by EudoraNiu** [comment]
Eudora's Rmd file for assignment 2. Thanks!

**Comment by EudoraNiu** [comment]
Thanks for the update Anna. I have uploaded the .Rmd file.

File: RAssignment 2 copy.Rmd [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+
+plot(g,layout=layout.fruchterman.reingold,vertex.size=(EDGE$count*3),edge.arrow.size=.4,edge.length=0.6,vertex.label.size=0.1)
+```
+
```
Comment (lizarova777) [inline RAssignment 2 copy.Rmd]: You are supposed to modify the vertex dataframe.

File: RAssignment 2 copy.Rmd [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+DS1<-read.csv("hudk4050-classes.csv")
+library(tidyr)
+DS2<-unite(DS1, Student,First.Name:Last.Name)
+DS2$'Class.1'[34] <- "HUDK 4050"
+DS2$'Class.2'[34] <- "HUDM 4050"
+DS2$'Class.3'[34] <- "HUDM 5026"
+DS2$'Class.4'[34] <- "HUDM 5126"
```
Comment (lizarova777) [inline RAssignment 2 copy.Rmd]: You can also use the `gsub()` function.

File: RAssignment 2 copy.Rmd [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+DS4<-select(DS3, -ClassNumber)
+DS5<-table(DS4$Student,DS4$Class)
+M1<-as.matrix(DS5)
+M2<-t(M1)
+M3<-M1 %*% M2
+library(igraph)
+P1<-graph_from_adjacency_matrix(M3)
+P2<-plot(P1)
+centrality <- centr_degree(P1, mode = c("all"))
+P3 <- plot(graph_from_adj_list(centrality))
```
Comment (lizarova777) [inline RAssignment 2 copy.Rmd]: Great job overall! Find the network metrics and interpret them within the context of the dataset. In other words, what does it mean to be the most central person in this network?

**Pull request closed without merging**